pipeline/container/routes/v4/runs.py

```diff
@@ -31,9 +33,23 @@ async def run(
     request: Request,
     response: Response,
 ):
-    # run_manager: Manager = request.app.state.manager
-    # outputs = await run_manager.run(run_create.inputs)
-    # return outputs
+    manager: Manager = request.app.state.manager
+    if manager.pipeline_state == pipeline_schemas.PipelineState.loading:
+        logger.info("Pipeline loading")
+        return run_schemas.ContainerRunResult(
+            outputs=None,
+            error=run_schemas.ContainerRunError.pipeline_loading,
+            error_message="Pipeline is still loading",
+        )
+
+    if manager.pipeline_state == pipeline_schemas.PipelineState.failed:
+        logger.info("Pipeline failed to load")
+        return run_schemas.ContainerRunResult(
+            outputs=None,
+            error=run_schemas.ContainerRunError.startup_error,
+            error_message="Pipeline failed to load",
+            error_traceback=manager.pipeline_state_message,
```

Comment:
I wonder if we shouldn't use this error output exclusively for things that happen inside the pipeline? Like, our system did everything as expected, here are the results. your code broke, vs an endpoint returning a coded error when something is wrong like a pipeline is still loading or broke etc.

Reply:
yep agreed, but in this case it will be reporting errors that happened in the user's code when their pipeline starts up (any functions marked as `on_startup=True`)

Reply:
yeah. Paul implemented the container startup stuff so that the startup happened asynchronously. So people can still make runs even though the pipeline hasn't finished starting up